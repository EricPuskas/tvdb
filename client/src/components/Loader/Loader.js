import React from "react";
import Rolling from "../../assets/img/rolling.svg";
import "./Loader.scss";

const Loader = () => {
  return (
    <div className="Loader" data-test="loader">
      <img src={Rolling} data-test="rolling" alt="Loading..." />
      <h2> Loading... </h2>
    </div>
  );
};

export default Loader;
