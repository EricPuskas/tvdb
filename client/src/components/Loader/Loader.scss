.Loader {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  align-content: center;

  img {
    height: auto;
    display: block;
    max-width: 80px;
  }

  h2 {
    margin-top: 0.5rem;
    font-size: 1.3rem;
  }
}
