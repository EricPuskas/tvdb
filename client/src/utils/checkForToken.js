import jwt_decode from "jwt-decode";

// Redux Store
import store from "../store";

// Utilities
import setToken from "./setToken";

// Actions
import { getToken } from "../actions/token";

const checkForToken = () => {
  // Check for token
  if (!localStorage.token) {
    store.dispatch(getToken());
    setToken(localStorage.token);
  } else {
    setToken(localStorage.token);
    // Decode token to be able to check if it expired
    const decoded = jwt_decode(localStorage.token);
    // Check for expired token
    const currentTime = Date.now() / 1000;
    if (decoded.exp < currentTime) {
      store.dispatch(getToken());
    }
  }
};

export default checkForToken;
