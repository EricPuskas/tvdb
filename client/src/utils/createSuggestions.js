const filterInput = input => {
  return input.replace(/[^\w\s]/gi, "");
};

const createSuggestions = (word, data) => {
  let filter = filterInput(word);
  const re = new RegExp(`${filter.toLowerCase()}.*\\B`, "g");
  return data.filter(item => re.test(item.toLowerCase()));
};

export default createSuggestions;
