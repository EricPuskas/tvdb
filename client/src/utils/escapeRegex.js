const escapeRegex = text => {
  if (text !== "" || text !== null || text !== undefined) {
    // replace all white space with +, globally (replace all occurrences)
    return text.replace(/[\s]/g, "+");
  }
};

export default escapeRegex;
