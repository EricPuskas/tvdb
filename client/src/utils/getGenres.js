import React from "react";
import uuid from "uuid/v4";

const getGenres = genres => {
  if (!genres) return null;
  return genres.map(genre => (
    <li key={uuid()}>
      <span className="tag">{genre}</span>
    </li>
  ));
};

export default getGenres;
