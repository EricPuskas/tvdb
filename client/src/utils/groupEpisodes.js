const groupEpisodes = data => {
  let seasons = [];
  for (let i = 0; i < data.length; i++) {
    if (!seasons[data[i].airedSeason]) seasons[data[i].airedSeason] = [];
    seasons[data[i].airedSeason].push({
      id: data[i].id,
      season: data[i].airedSeason,
      episode: data[i].airedEpisodeNumber,
      title: data[i].episodeName
    });
  }
  return seasons;
};

export default groupEpisodes;
