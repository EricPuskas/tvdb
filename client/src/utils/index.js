import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
