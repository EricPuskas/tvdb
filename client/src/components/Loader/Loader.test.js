import React from "react";
import { shallow } from "enzyme";
import { findByAttribute } from "../../utils";
import Loader from "./Loader";

describe("Loader Component", () => {
  const component = shallow(<Loader />);
  const wrapper = findByAttribute(component, "loader");
  const image = findByAttribute(component, "rolling");

  it("Should render without errors", () => {
    expect(wrapper.length).toBe(1);
  });

  it("Should render spinner without errors", () => {
    expect(image.length).toBe(1);
  });

  it("Should pass snapshot test", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
