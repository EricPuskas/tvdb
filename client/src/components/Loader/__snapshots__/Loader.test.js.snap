// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Should pass snapshot test 1`] = `"<div class=\\"Loader\\" data-test=\\"loader\\"><img src=\\"rolling.svg\\" data-test=\\"rolling\\" alt=\\"Loading...\\"/><h2> Loading... </h2></div>"`;
